src/number_theory/bertrand.lean

```diff
@@ -0,0 +1,1559 @@
+  cases multiplicity_pos with m hm,
+  simp only [finset.Ico.mem, finset.mem_filter] at hm,
+  calc p = p ^ 1 : tactic.ring_exp.base_to_exp_pf rfl
+  ...    ≤ p ^ m : nat.pow_le_pow_of_le_right (by linarith [hp.out.one_lt]) hm.left.left
+  ...    ≤ 2 * (n % p ^ m) : hm.right
+  ...    ≤ 2 * n : nat.mul_le_mul_left _ (nat.mod_le n _),
+end
+
+/-
+Then:
+4^n ≤ 2nCn * (2 * n + 1) (by choose_halfway_is_big)
+= prod (primes ≤ 2n) p^(α n p) * (2n+1)
+= prod (primes ≤ n) p^(α n p) * prod (primes n < p <= 2n) p^α * (2n+1)
+= prod (primes ≤ 2n/3) p^α * prod (primes 2n/3 to n) p^α * prod (primes n < p ≤ 2n) p^α * (2n+1)
+= prod (primes ≤ 2n/3) p^α * prod (primes 2n/3 to n) 1 * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by claim 3)
+= prod (primes ≤ 2n/3) p^α * prod (primes n < p ≤ 2n) p^α * (2n+1)
+= prod (primes ≤ sqrt(2n)) p^α * prod(primes sqrt(2n)..2n/3) p^α
+  * prod (primes n < p ≤ 2n) p^α * (2n+1)
+≤ prod (primes ≤ sqrt(2n)) p^α * prod(primes sqrt(2n)..2n/3) p
+  * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by claim 2)
+≤ prod (primes ≤ sqrt(2n)) p^α * 4 ^ (2n / 3) * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by a general bound on the primorial)
+≤ prod (primes ≤ sqrt(2n)) (2n) * 4 ^ (2n / 3) * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by claim 1)
+= (2n)^π (sqrt 2n) * 4 ^ (2n/3) * prod (primes n < p ≤ 2n) p^α * (2n+1)
+≤ (2n)^(sqrt 2n) * 4 ^ (2n/3) * prod (primes n < p ≤ 2n) p^α * (2n+1)
+  (by "prime count of x is less than x")
+
+For sufficiently large n, that last product term is > 1.
+Indeed, suppose for contradiction it's equal to 1.
+Then 4^n ≤ (2n)^(sqrt 2n) * 4^(2n/3) * (2n+1)
+so 4^(n/3) ≤ (2n)^(sqrt 2n) (2n+1)
+and this is Clearly False for sufficiently large n.
+-/
+
+lemma two_n_div_3_le_two_mul_n_choose_n (n : ℕ) : 2 * n / 3 < (2 * n).choose n :=
+begin
+  cases n,
+  { simp only [nat.succ_pos', nat.choose_self, nat.zero_div, mul_zero], },
+  calc 2 * (n + 1) / 3 < 2 * (n + 1): nat.div_lt_self (by norm_num) (by norm_num)
+  ... = (2 * (n + 1)).choose(1): by norm_num
+  ... ≤ (2 * (n + 1)).choose(2 * (n + 1) / 2): nat.choose_le_middle 1 (2 * (n + 1))
+  ... = (2 * (n + 1)).choose(n + 1): by simp only [nat.succ_pos', nat.mul_div_right]
+end
+
+lemma not_pos_iff_zero (n : ℕ) : ¬ 0 < n ↔ n = 0 :=
+begin
+  split,
+  { intros h, induction n, refl, simp only [nat.succ_pos', not_true] at h, cc, },
+  { intros h, rw h, exact irrefl 0, },
+end
+
+lemma alskjhads (n x : ℕ): 2 * n / 3 + 1 ≤ x -> 2 * n < 3 * x :=
+begin
+  intro h,
+  rw nat.add_one_le_iff at h,
+  cases lt_or_ge (2 * n) (3 * x),
+  { exact h_1, },
+  { exfalso,
+    simp only [ge_iff_le] at h_1,
+    induction x,
+    { simp at h, exact h, },
+    { apply x_ih,
+      cases lt_or_ge (2 * n / 3) x_n,
+      { exact h_2, },
+      { have r : 2 * n / 3 = x_n, by omega,
+        exfalso,
+        subst r,
+        exact nat.lt_le_antisymm (nat.lt_mul_div_succ (2 * n) (by norm_num)) h_1, },
+      { calc 3 * x_n ≤ 3 * (x_n + 1): by norm_num
+        ... ≤ 2 * n: h_1, }, }, },
+end
```

Comment:
```suggestion
lemma alskjhads_no_two (n x : ℕ) (h : n / 3 + 1 ≤ x) : n < 3 * x :=
lt_of_lt_of_le ((nat.div_lt_iff_lt_mul' zero_lt_three).mp (nat.succ_le_iff.mp h)) (mul_comm _ _).le

lemma alskjhads (n x : ℕ): 2 * n / 3 + 1 ≤ x -> 2 * n < 3 * x :=
alskjhads_no_two (2 * n) x
```
In the version with which I was working, there was a missing `ordered_comm_semiring` instance on `nat`.  With that instance, the above works.

You may not even need the exact version that you proved: you should be able to simply apply the `not_two`-version to `2 * n`.